tests/tests_integration/test_api/test_sequences.py

```diff
@@ -329,3 +334,58 @@ def test_aggregate_unique_metadata_keys(self, cognite_client: CogniteClient, seq
         assert {tuple(item.value["property"]) for item in result} >= {
             ("metadata", key.casefold()) for a in sequence_list for key in a.metadata or []
         }
+
+    def test_upsert_sequence(self, cognite_client: CogniteClient) -> None:
+        original_sequence = SequenceWrite(
+            external_id=f"upsert_sequence_{random_string(5)}",
+            columns=[
+                SequenceColumnWrite(
+                    description="KW Description", name="KW Name", value_type="Double", external_id="kw_seq_01"
+                ),
+            ],
+            description="Description of the Test Sequence",
+            name="Test Sequence Name",
+        )
+        upsert = SequenceWrite(
+            external_id=original_sequence.external_id,
+            columns=[
+                SequenceColumnWrite(
+                    description="KW Description",
+                    name="KW Name",
+                    # UPPER to match what the API returns
+                    value_type="DOUBLE",
+                    external_id="kw_seq_01",
+                    metadata={},
+                ),
+                SequenceColumnWrite(
+                    description="PW Description",
+                    name="PW Name",
+                    value_type="DOUBLE",
+                    external_id="pw_seq_01",
+                    metadata={},
+                ),
+                SequenceColumnWrite(
+                    description="LW Description",
+                    name="LW Name",
+                    value_type="DOUBLE",
+                    external_id="lw_seq_01",
+                    metadata={},
+                ),
+            ],
+            description="Description of the Test Sequence",
+            name="Test Sequence Name",
+        )
+
+        created: Sequence | None = None
+        try:
+            created = cognite_client.sequences.create(original_sequence)
+
+            upserted = cognite_client.sequences.upsert(upsert)
+
+            retrieved = cognite_client.sequences.retrieve(external_id=upserted.external_id)
+
+            assert retrieved is not None
+            assert retrieved.as_write().columns.dump() == upsert.columns.dump()
```

Comment:
This test implicitly uses `mode="replace_ignore_null"`, we should ensure `"replace"` and `"patch"` also works as expected